tests/mortar/tests.h

```diff
@@ -470,3 +470,323 @@ class StokesOperator
+                        const DoFHandler<dim>           &dof_handler,
+                        const AffineConstraints<Number> &constraints,
+                        const Quadrature<dim>           &quadrature,
+                        const double                     delta_1_scaling)
+    : delta_1_scaling(delta_1_scaling)
+  {
+    reinit(mapping, dof_handler, constraints, quadrature);
+  }
+
+  void
+  reinit(const Mapping<dim>              &mapping,
+         const DoFHandler<dim>           &dof_handler,
+         const AffineConstraints<Number> &constraints,
+         const Quadrature<dim>           &quadrature)
+  {
+    typename MatrixFree<dim, Number, VectorizedArrayType>::AdditionalData data;
+    data.mapping_update_flags =
+      update_quadrature_points | update_gradients | update_values;
+
+    matrix_free.reinit(mapping, dof_handler, constraints, quadrature, data);
+
+    valid_system = false;
+  }
+
+  /**
+   * @brief Create coupling operator
+   */
+  void
+  add_coupling(const unsigned int n_subdivisions,
+               const double       radius,
+               const double       rotate_pi,
+               const unsigned int bid_0,
+               const unsigned int bid_1,
+               const double       sip_factor = 1.0)
+  {
+    coupling_operator_v = std::make_shared<CouplingOperator<dim, dim, Number>>(
+      *matrix_free.get_mapping_info().mapping,
+      matrix_free.get_dof_handler(),
+      matrix_free.get_affine_constraints(),
+      matrix_free.get_quadrature(),
+      n_subdivisions,
+      radius,
+      rotate_pi,
+      bid_0,
+      bid_1,
+      sip_factor,
+      0);
+    coupling_operator_p = std::make_shared<CouplingOperator<dim, 1, Number>>(
+      *matrix_free.get_mapping_info().mapping,
+      matrix_free.get_dof_handler(),
+      matrix_free.get_affine_constraints(),
+      matrix_free.get_quadrature(),
+      n_subdivisions,
+      radius,
+      rotate_pi,
+      bid_0,
+      bid_1,
+      sip_factor,
+      dim);
+  }
+
+  virtual types::global_dof_index
+  m() const
+  {
+    if (this->matrix_free.get_mg_level() != numbers::invalid_unsigned_int)
+      return this->matrix_free.get_dof_handler().n_dofs(
+        this->matrix_free.get_mg_level());
+    else
+      return this->matrix_free.get_dof_handler().n_dofs();
+  }
+
+  Number
+  el(unsigned int, unsigned int) const
+  {
+    DEAL_II_NOT_IMPLEMENTED();
+    return 0;
+  }
+
+  void
+  initialize_dof_vector(VectorType &dst) const
+  {
+    matrix_free.initialize_dof_vector(dst);
+  }
+
+  void
+  vmult(VectorType &dst, const VectorType &src) const
+  {
+    src.update_ghost_values();
+
+    matrix_free.cell_loop(
+      &GeneralStokesOperator<dim, Number, VectorizedArrayType>::do_vmult_cell,
+      this,
+      dst,
+      src,
+      true);
+
+    // apply coupling terms
+    if (coupling_operator_v)
+      coupling_operator_v->vmult_add(dst, src);
+
+    if (coupling_operator_p)
+      coupling_operator_p->vmult_add(dst, src);
```

Comment:
For now, I have split up the coupling operator into two parts (one for pressure and one for velocity) so that I will be to handle different polynomial degrees.